kaldb/src/main/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumer.java

```diff
@@ -287,31 +288,41 @@ public boolean consumeMessagesBetweenOffsetsInParallel(
         messagesIndexed += recordCount;
         executor.execute(
             () -> {
-              LOG.info("Ingesting batch: [{}/{}]", topicPartition, recordCount);
-              for (ConsumerRecord<String, byte[]> record : records) {
-                if (startOffsetInclusive >= 0 && record.offset() < startOffsetInclusive) {
-                  messagesOutsideOffsetRange.incrementAndGet();
-                  recordsFailedCounter.increment();
-                } else if (endOffsetInclusive >= 0 && record.offset() > endOffsetInclusive) {
-                  messagesOutsideOffsetRange.incrementAndGet();
-                  recordsFailedCounter.increment();
-                } else {
-                  try {
-                    if (logMessageWriterImpl.insertRecord(record)) {
-                      recordsReceivedCounter.increment();
-                    } else {
-                      recordsFailedCounter.increment();
+              long startTime = System.nanoTime();
+              UUID batchUuid = UUID.randomUUID();
```

Comment:
Adds a batch UUID to uniquely identify log entries for this batch. Previously, the 3-tuple of [topic, partition, batch record count] was used and that 3-tuple is very much not unique.

Reply:
I think dropping the uuid would be ideal since beyond the scope of this function, the uuid has no utility. Further, it would be good to any extra work on ingestion.

Reply:
It's easy to drop the uuid, but its value comes from the fact that right now we have a bunch of interleaved log messages that look like this:

```
Ingesting batch: [kaldb_preprocessor-1/500]
Ingesting batch: [kaldb_preprocessor-1/500]
Finished ingesting batch: [kaldb_preprocessor-1/500]
Ingesting batch: [kaldb_preprocessor-1/500]
Finished ingesting batch: [kaldb_preprocessor-1/500]
Finished ingesting batch: [kaldb_preprocessor-1/500]
```

We can't identify the "Finished" log message that corresponds to an "Ingesting batch" message because `topic-partionId/recordCount` doesn't uniquely identify a batch. Is that useful or do you still want the UUID dropped?

Reply:
Can we use the thread-id instead to know the start stop instead of a new uuid?

Reply:
:+1: